src/azure-cli/azure/cli/command_modules/vm/_help.py

```diff
@@ -2138,8 +2138,14 @@
 
 helps['vm secret add'] = """
 type: command
-short-summary: Add a secret to a VM.
+short-summary: Add a secret to a VM. To install certificates on a virtual machine it is recommended to use the [Azure Key Vault virtual machine extension for Linux](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-linux) or the [Azure Key Vault virtual machine extension for Windows](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-windows) instead of `az vm secret add`.
 examples:
+  - name: Add a secret to a VM using the Azure Key Vault virtual machine extension (Linux).
+    text: |
+        az vm extension add -n "KeyVaultForLinux" --publisher Microsoft.Azure.KeyVault -g "<resourcegroup>" --vm-name "<vmName>" --version 2.0 --settings '{\"secretsManagementSettings\": { \"pollingIntervalInS\": \"<pollingInterval>\", \"certificateStoreName\": \"<certStoreName>\", \"certificateStoreLocation\": \"<certStoreLoc>\", \"observedCertificates\": [\" <observedCert1> \", \" <observedCert2> \"] }}'
```

Comment:
```suggestion
        az vm secret add -n "KeyVaultForLinux" --publisher Microsoft.Azure.KeyVault -g "<resourcegroup>" --vm-name "<vmName>" --version 2.0 --settings '{\"secretsManagementSettings\": { \"pollingIntervalInS\": \"<pollingInterval>\", \"certificateStoreName\": \"<certStoreName>\", \"certificateStoreLocation\": \"<certStoreLoc>\", \"observedCertificates\": [\" <observedCert1> \", \" <observedCert2> \"] }}'
```

Reply:
Are you sure that's correct? In the product docs we have the sample as az vm extension add

https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/key-vault-windows#azure-cli-deployment

![image](https://user-images.githubusercontent.com/32313503/126667683-94c15a76-7563-40e9-a164-d939d370f9bd.png)

Reply:
Please note that this command declared in help is `vm secret add`. The example for command should be consistent with the command declared in help
![Screenshot 2021-07-23 101209](https://user-images.githubusercontent.com/16612065/126730216-656e96a3-a068-4268-83f3-6b673584e252.png)

Reply:
I assume this example should be added in the wrong position, and it should be added under `az vm extension add`. You may need to add a new help entry for `vm extension add`

Reply:
We wanted to add it here because it is more secure to add a secret using the extension then the add secret command. Basically directing customers to use the most secure way possible. 

@smotwani what do you think?

Reply:
In fact, adding the example of other commands here will make users feel that the example does not match the command. And in this way, this PR cannot pass the check of CI Linter.


If you want to guide users to use the new command, I recommend you add `c.deprecate()` flag to the original command. In this way, the deprecated information will be prompted in help.
Such as:
```
 with self.command_group('group deployment', resource_deployment_sdk, deprecate_info=self.deprecate(redirect='deployment group', hide=True)) as g:
```
The deprecated information in help
![Screenshot 2021-07-24 122223](https://user-images.githubusercontent.com/16612065/126857214-de697bcb-db26-483b-80f2-e244de361a8a.png)

For more details, please refer to [deprecating-commands-and-arguments](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments)